### PR TITLE
add http service

### PR DESCRIPTION
**Pull request by SayPls**

### CODE REVIEW

File: frontend/package-lock.json [hunk 1]
```diff
@@ -1,8 +1,11288 @@
 {
```
Comment (ismv) [inline frontend/package-lock.json]: этот файл коммитить не нужно

File: frontend/src/app/core/services/codi-http-client.service.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    if (url.startsWith('http://') || url.startsWith('https://')) {
+      return url;
+    }
+    return this.baseUrl + url;
+  }
+  public prepareData(payload: object): string {
+    return JSON.stringify(payload);
+  }
+
+  private handleError(error: HttpErrorResponse){
```
Comment (ismv) [inline frontend/src/app/core/services/codi-http-client.service.ts:64]: add explicit return type here

File: frontend/src/app/core/services/codi-http-client.service.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    if(error.error instanceof ErrorEvent){
+      //error client
+      errorMessage = error.error.message;
+    } else {
+      //error server
+      errorMessage = `Error: ${error.status}, ` + `message: ${error.message}`;
+    }
+
+    return throwError(errorMessage);
```
[the remark this answers is not in the document]
Comment (SayPls) [reply frontend/src/app/core/services/codi-http-client.service.ts:75]: > throwError(err) is deprecated, use throwError(() => err) instead

return throwError(() => errorMessage);

File: .gitignore [hunk 4]
```diff
@@ -9,6 +9,7 @@
 *.user
 *.userosscache
 *.sln.docstates
+*.idea
```
Comment (tatianahutii) [inline .gitignore:12]: Why did you add this?

File: backend/.idea/.idea.Codi.All/.idea/.gitignore [hunk 5]
```diff
@@ -0,0 +1,13 @@
+﻿# Default ignored files
```
Comment (tatianahutii) [inline backend/.idea/.idea.Codi.All/.idea/.gitignore:1]: I believe we don't need these files

File: backend/.idea/.idea.Codi.All/.idea/.name [hunk 6]
```diff
@@ -0,0 +1 @@
+Codi.All
```
Comment (tatianahutii) [inline backend/.idea/.idea.Codi.All/.idea/.name:1]: See previous comment

File: backend/.idea/.idea.Codi.All/.idea/encodings.xml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tatianahutii) [inline backend/.idea/.idea.Codi.All/.idea/encodings.xml:1]: See previous comment

File: backend/.idea/.idea.Codi.All/.idea/indexLayout.xml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tatianahutii) [inline backend/.idea/.idea.Codi.All/.idea/indexLayout.xml:1]: backend/.idea/.idea.Codi.All/.idea/indexLayout.xml

File: backend/.idea/.idea.Codi.All/.idea/vcs.xml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tatianahutii) [inline backend/.idea/.idea.Codi.All/.idea/vcs.xml:1]: backend/.idea/.idea.Codi.All/.idea/indexLayout.xml

**Review by tatianahutii — CHANGES_REQUESTED** [review]
